.github/workflows/test.yaml

```diff
@@ -0,0 +1,33 @@
+name: Test
+
+on:
+  pull_request:
+    branches: [ 'main' ]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Test Projects
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: step-security/harden-runner@4d991eb9b905ef189e4c376166672c3f2f230481 # v2.11.0
+      with:
+        egress-policy: audit
+
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Set up Go
+      uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
+      with:
+        go-version-file: './tw/go.mod'
+        cache-dependency-path: '**/*.sum'
+
+    - name: Test all modules
+      run: |
+        for dir in $(find . -name "go.mod" -exec dirname {} \;); do
+          echo "Testing in $dir"
+          make -C "$dir" test
+        done
```

Comment:
```bash
#!/bin/sh
set -e
for dir in $(false); do
   echo "false should have output something: $dir"
done
echo "great, everything looks dandy"
exit 0

```

Then 
```
$ bash example.sh 
great, everything looks dandy
```

Reply:
the point there is that the `find . ` just swallows errors like that.

Reply:
I took another crack and just leaned into make, I stubb'd in a `test` target everywhere and discovered the project targets in the root make

Reply:
notably this will fail if no `test` target is found in a project, but I think thats WAI